src/Simple/JSON.purs

```diff
@@ -96,3 +103,33 @@ instance readFieldsNil ::
   ) => ReadForeignFields Nil row where
   getFields _ _ _ =
     pure $ to {}
+
+
+-- | The purpose of the `ReadKey` class is to turn strings into map keys,
+-- | so we can read generic `Map`s from `StrMap`s.
+class ReadKey a where
+  readKey :: String -> F a
```

Comment:
huh... i also removed the note about not implementing typeclasses of the main class though, since we already don't have complete compatibility with the raw JSON form. i think it'll be fine to use either the simple JSON or foreign-generics Decode instances for those, since people can use the sum decoding from foreign-generics for sums (and will need them for any fairly complicated keys)